src/umzug.ts

```diff
@@ -483,6 +480,30 @@ export class Umzug<Ctx extends object = object> extends emittery<UmzugEvents<Ctx
 		});
 	}
 
+	private async getContext(): Promise<Ctx> {
+		const isAsyncFunc = (_ctx: object | Function) => {
```

Comment:
This seems brittle and not necessary. If it's a sync or async function we can do `await fn()` to gets the result

Reply:
You're right, thanks. I've got checked it out. Please review.